src/main/java/org/openrewrite/java/spring/boot3/RemoveSolrAutoConfigurationExclude.java

```diff
@@ -0,0 +1,95 @@
+    private static final AnnotationMatcher EAC_MATCHER = new AnnotationMatcher(ENABLE_AUTO_CONFIGURATION);
+
+    private static final String SOLR_AUTO_CONFIGURATION = "SolrAutoConfiguration";
+    private static final String SOLR_AUTOCONFIGURATION_FQN = "org.springframework.boot.autoconfigure.solr." + SOLR_AUTO_CONFIGURATION;
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(Preconditions.or(
+                        new UsesType<>(SPRING_BOOT_APPLICATION, true),
+                        new UsesType<>(ENABLE_AUTO_CONFIGURATION, true)),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.Annotation visitAnnotation(J.Annotation annotation, ExecutionContext ctx) {
+                        J.Annotation a = super.visitAnnotation(annotation, ctx);
+                        if (!SBA_MATCHER.matches(a) && !EAC_MATCHER.matches(a)) {
+                            return a;
+                        }
+                        maybeRemoveImport(SOLR_AUTOCONFIGURATION_FQN);
+
+                        return a.withArguments(ListUtils.map(a.getArguments(), it -> {
+                            if (it instanceof J.Assignment) {
+                                J.Assignment as = (J.Assignment) it;
+                                if (as.getAssignment() == null || !"exclude".equals(((J.Identifier) as.getVariable()).getSimpleName())) {
+                                    return it;
+                                }
+                                if (isSolrAutoConfigurationClassReference(as.getAssignment())) {
+                                    return null;
+                                } else if (as.getAssignment() instanceof J.NewArray) {
+                                    J.NewArray array = (J.NewArray) as.getAssignment();
+                                    List<Expression> newInitializer = ListUtils.map(array.getInitializer(),
+                                            expr -> isSolrAutoConfigurationClassReference(expr) ? null : expr);
+                                    //noinspection DataFlowIssue
+                                    if (newInitializer.isEmpty()) {
+                                        return null;
+                                    }
+                                    return maybeAutoFormat(it, as.withAssignment(array.withInitializer(newInitializer)), ctx);
+                                }
+                            }
+                            return it;
+                        }));
+                    }
+
+                    private boolean isSolrAutoConfigurationClassReference(Expression expr) {
+                        return expr instanceof J.FieldAccess &&
+                               TypeUtils.isAssignableTo(SOLR_AUTOCONFIGURATION_FQN, ((J.FieldAccess) expr).getTarget().getType());
```

Comment:
@Laurens-W A pattern I've seen you use a couple times now is to compare with `.equals("SomeString")`. Try to use `TypeUtils` where you can for type safe comparisons instead, as that then also handles the FQN case as seen in the added unit test.